ffs/scheduler/scheduler_import.go

```diff
@@ -0,0 +1,124 @@
+package scheduler
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/ipfs/go-cid"
+	"github.com/textileio/powergate/ffs"
+	"github.com/textileio/powergate/ffs/scheduler/internal/cistore"
+)
+
+// ImportDeals create or augments a Storageinfo for a Cid with the provided deal ids.
+// All deal ids must be active on-chain to let the operation succeed.
+func (s *Scheduler) ImportDeals(iid ffs.APIID, payloadCid cid.Cid, dealIDs []uint64) error {
+	// 1. Get current StorageInfo.
+	si, err := s.cis.Get(iid, payloadCid)
+	if err == cistore.ErrNotFound {
+		si = ffs.StorageInfo{
+			APIID:   iid,
+			JobID:   ffs.EmptyJobID,
+			Cid:     payloadCid,
+			Created: time.Now(),
+			Cold: ffs.ColdInfo{
+				Filecoin: ffs.FilInfo{
+					DataCid: payloadCid,
+				},
+			},
+		}
+	} else if err != nil {
+		return fmt.Errorf("getting current storageinfo: %s", err)
+	}
+
+	// 2. Augment the retrieved/bootstraped StorageInfo with provided deals.
+	if err := s.augmentStorageInfo(&si.Cold.Filecoin, dealIDs); err != nil {
+		return fmt.Errorf("generating storage info from imported deals: %s", err)
+	}
+	if err := s.cis.Put(si); err != nil {
+		return fmt.Errorf("importing cid information: %s", err)
+	}
+
+	return nil
+}
+
+func (s *Scheduler) augmentStorageInfo(fi *ffs.FilInfo, deals []uint64) error {
+	// Deduplicate any already known DealID existing in StorageInfo with the
+	// imported ones.
+	var newDealIDs []uint64
+	existingDealID := map[uint64]struct{}{}
+	for _, p := range fi.Proposals {
+		existingDealID[p.DealID] = struct{}{}
+	}
+	for _, did := range deals {
+		if _, ok := existingDealID[did]; !ok {
+			newDealIDs = append(newDealIDs, did)
+		}
+	}
```

Comment:
There's the possibility that the user is providing DealIDs that we already know. Just be sure to take them out for the importing logic.